### PR TITLE
Add per-notification setting to toggle window highlight (if not active)

### PR DESCRIPTION
**Pull request by davidebeatrici**
As requested in #3283.

### CODE REVIEW

**Comment by davidebeatrici** [comment]
As a global setting or for each notification?

**Comment by mkrautz** [comment]
Global setting.

**Comment by davidebeatrici** [comment]
Should it be in the **User Interface** or **Messages** tab?

**Comment by mkrautz** [comment]
Actually, it'd probably make most sense as a per-notification setting. People probably want to get Mumble highlighted for some notifications, not others. Text messages would be a good example.

How about calling it "Highlight"?

**Comment by davidebeatrici** [comment]
I agree, having a per-notification setting would be perfect.

"Highlight" sounds good, with a tooltip explaining what it means in detail.

**Comment by davidebeatrici** [comment]
Done.

**Comment by davidebeatrici** [comment]
Done.

@bontibon @mkrautz

File: src/mumble/Log.cpp [hunk 1]
```diff
@@ -54,15 +56,18 @@ LogConfig::LogConfig(Settings &st) : ConfigWidget(st) {
 		twi->setText(ColMessage, messageName);
 		twi->setCheckState(ColConsole, Qt::Unchecked);
 		twi->setCheckState(ColNotification, Qt::Unchecked);
+		twi->setCheckState(ColHighlight, Qt::Unchecked);
 		twi->setCheckState(ColStaticSound, Qt::Unchecked);
 
 		twi->setToolTip(ColConsole, tr("Toggle console for %1 events").arg(messageName));
 		twi->setToolTip(ColNotification, tr("Toggle pop-up notifications for %1 events").arg(messageName));
+		twi->setToolTip(ColHighlight, tr("Toggle window highlight in taskbar (if not active) for %1 events").arg(messageName));
```
Comment (mkrautz) [inline src/mumble/Log.cpp]: Taskbar is not a cross-platform term.

File: src/mumble/Log.cpp [hunk 2]
```diff
@@ -54,15 +56,18 @@ LogConfig::LogConfig(Settings &st) : ConfigWidget(st) {
...
 		twi->setToolTip(ColStaticSound, tr("Click here to toggle sound notification for %1 events").arg(messageName));
 		twi->setToolTip(ColStaticSoundPath, tr("Path to sound file used for sound notifications in the case of %1 events<br />Single click to play<br />Double-click to change").arg(messageName));
 
 		twi->setWhatsThis(ColConsole, tr("Click here to toggle console output for %1 events.<br />If checked, this option makes Mumble output all %1 events in its message log.").arg(messageName));
 		twi->setWhatsThis(ColNotification, tr("Click here to toggle pop-up notifications for %1 events.<br />If checked, a notification pop-up will be created by Mumble for every %1 event.").arg(messageName));
+		twi->setWhatsThis(ColHighlight, tr("Click here to toggle window highlight in taskbar for %1 events.<br />If checked, Mumble's window will be highlighted in the taskbar for every %1 event, if not active.").arg(messageName));
```
Comment (mkrautz) [inline src/mumble/Log.cpp]: Taskbar is not a cross-platform term.

File: src/mumble/Log.cpp [hunk 3]
```diff
@@ -491,9 +499,17 @@ void Log::log(MsgType mt, const QString &console, const QString &terse, bool own
 	if (!g.s.bTTSMessageReadBack && ownMessage)
 		return;
 
-	// Message notification with balloon tooltips
-	if ((flags & Settings::LogBalloon) && !(g.mw->isActiveWindow() && g.mw->qdwLog->isVisible()))
-		postNotification(mt, plain);
+	if (!(g.mw->isActiveWindow() && g.mw->qdwLog->isVisible())) {
+		// Message notification with window highlight in taskbar
```
Comment (mkrautz) [inline src/mumble/Log.cpp]: Taskbar is not a cross-platform term.

File: src/mumble/Log.cpp (hunk 1 above)
Comment (davidebeatrici) [reply src/mumble/Log.cpp]: What about "Application bar"?

Comment (mkrautz) [reply src/mumble/Log.cpp]: How about just dropping "in taskbar"? Same for the rest of these.

Comment (davidebeatrici) [reply src/mumble/Log.cpp]: Sounds good.

Every desktop environment has its own way to highlight the window (if supported).

**Pull request merged**